src/pages/index.js

```diff
@@ -1,107 +1,244 @@
+          saveButton.textContent = "Save";
+        });
     }
   );
   profileEditPopup.setEventListeners();
 
-  // PopupWithForm for adding a new card
   const addCardPopup = new PopupWithForm("#profile-add-modal", (formData) => {
     const cardData = {
       name: formData["add-title"],
       link: formData["link-type"],
     };
-    const cardElement = createCard(cardData); // Create the new card
-    cardSection.addItem(cardElement); // Add it to the DOM
-    addCardPopup.close(); // Close the modal after adding the card
 
-    addCardForm.reset(); // Reset the form fields only after successful submission
-    addCardFormValidator.disableButton(); // Disable the button after form submission
+    // Change the button text to "Saving..."
+    const saveButton = document.querySelector("#modal-submit-button");
+    saveButton.textContent = "Saving...";
+
+    console.log("Sending card data:", cardData); // Log the card data being sent
+
+    api
+      .createCard(cardData)
+      .then((newCard) => {
+        console.log("Card added successfully:", newCard); // Log successful response
+        const cardElement = createCard(newCard, api); // Pass api to Card
+        cardSection.addItem(cardElement);
+        addCardPopup.close();
+        addCardForm.reset();
+        addCardFormValidator.disableButton();
+      })
+      .catch((error) => {
+        console.error("Error adding new card:", error); // Log any errors
+      })
+      .finally(() => {
+        saveButton.textContent = "Create"; // Reset the button text after the action
+      });
   });
   addCardPopup.setEventListeners();
 
-  /* ---------------------------- Form Validation ----------------------------- */
-  // Initialize form validators for both forms
+  const deleteConfirmationPopup = new PopupWithConfirmation(
+    "#delete-confirmation-modal",
+    (cardId) => {
+      api
+        .deleteCard(cardId)
+        .then(() => {
+          const cardElement = document.querySelector(`[data-id='${cardId}']`);
+          if (cardElement) {
+            cardElement.remove();
+          }
+          console.log("Card deleted successfully.");
+        })
+        .catch((error) => {
+          console.error("Error deleting card:", error);
+        });
+    }
+  );
+
   const profileFormValidator = new FormValidator(settings, profileEditForm);
   const addCardFormValidator = new FormValidator(settings, addCardForm);
-  // Enable form validation
   profileFormValidator.enableValidation();
   addCardFormValidator.enableValidation();
 
-  /* ---------------------------- Handlers ----------------------------- */
-  // Function to create a new Card instance
-  function createCard(cardData) {
-    const card = new Card(cardData, "#card-template", handleImageClick);
-    return card.getView();
+  // Select avatar form and initialize its validator
+  const avatarForm = document.querySelector("#profile-avatar-form");
+  const avatarFormValidator = new FormValidator(settings, avatarForm);
```

Comment:
**COULD BE IMPROVED**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```